packages/components/pie-checkbox/src/checkbox.scss

```
@@ -1 +1,276 @@
+                }
+            }
+        }
+
+        &:not([data-pie-disabled]) {
+            .c-checkbox-tick {
+                --checkbox-bg: var(--dt-color-interactive-brand);
+                --checkbox-border-color: var(--dt-color-interactive-brand);
+            }
+        }
+
+        &[data-pie-status="error"] {
+            .c-checkbox-tick {
+                --checkbox-bg: var(--dt-color-support-error);
+                --checkbox-border-color: var(--dt-color-support-error);
+            }
+        }
+
+        &:hover:not([data-pie-disabled]) {
+            .c-checkbox-tick {
+                --checkbox-bg: hsl(var(--dt-color-interactive-brand-h), var(--dt-color-interactive-brand-s), calc(var(--dt-color-interactive-brand-l) + calc(-1 * var(--dt-color-hover-01))));
+                --checkbox-border-color: hsl(var(--dt-color-interactive-brand-h), var(--dt-color-interactive-brand-s), calc(var(--dt-color-interactive-brand-l) + calc(-1 * var(--dt-color-hover-01))));
+            }
+        }
+
+        &:hover[data-pie-status="error"] {
+            .c-checkbox-tick {
+                --checkbox-bg: hsl(var(--dt-color-support-error-h), var(--dt-color-support-error-s), calc(var(--dt-color-support-error-l) + calc(-1 * var(--dt-color-hover-01))));
+                --checkbox-border-color: hsl(var(--dt-color-support-error-h), var(--dt-color-support-error-s), calc(var(--dt-color-support-error-l) + calc(-1 * var(--dt-color-hover-01))));
+            }
+        }
+
+        &:active:not([data-pie-disabled]) {
+            .c-checkbox-tick {
+                --checkbox-bg: hsl(var(--dt-color-interactive-brand-h), var(--dt-color-interactive-brand-s), calc(var(--dt-color-interactive-brand-l) + calc(-1 * var(--dt-color-active-01))));
+                --checkbox-border-color: hsl(var(--dt-color-interactive-brand-h), var(--dt-color-interactive-brand-s), calc(var(--dt-color-interactive-brand-l) + calc(-1 * var(--dt-color-active-01))));
+            }
+        }
+
+        &:active[data-pie-status="error"] {
+            .c-checkbox-tick {
+                --checkbox-bg: hsl(var(--dt-color-support-error-h), var(--dt-color-support-error-s), calc(var(--dt-color-support-error-l) + calc(-1 * var(--dt-color-active-01))));
+                --checkbox-border-color: hsl(var(--dt-color-support-error-h), var(--dt-color-support-error-s), calc(var(--dt-color-support-error-l) + calc(-1 * var(--dt-color-active-01))));
+            }
+        }
+    }
+
+    // styles for indeterminate state
+    .c-checkbox-tick:after {
+        content: "";
+
+        position: relative;
+        top: 47%;
+        left: 14%;
+        width: 0;
+        height: 2px;
+        background-color: white;
+    }
+
+    &[data-pie-indeterminate][data-pie-checked] {
+        .c-checkbox-tick:after {
+            width: 16px;
+
+            transition: width var(--dt-motion-timing-150) var(--checkbox-motion-easing);
+        }
+    }
```

I'd move this out of the nesting so you can have:

```suggestion
    // styles for indeterminate state
    .c-checkbox-tick:after {
        content: "";

        position: relative;
        top: 47%;
        left: 14%;
        width: 0;
        height: 2px;
        background-color: white;
        
       [data-pie-indeterminate][data-pie-checked] & {
           width: 16px;

            transition: width var(--dt-motion-timing-150) var(--checkbox-motion-easing);
       }
    }
```